sqlmesh/core/context.py

```diff
@@ -1830,6 +1830,14 @@ def create_external_models(self, strict: bool = False) -> None:
             external_models_yaml = (
                 path / c.EXTERNAL_MODELS_YAML if not deprecated_yaml.exists() else deprecated_yaml
             )
+
+            external_models_gateway: t.Optional[str] = self.gateway or self.config.default_gateway
```

Comment:
Should there be any specific logic around handling existing records with empty gateway?

Reply:
I couldn't come up with a test case that needed anything beyond [what already exists](https://github.com/TobikoData/sqlmesh/blob/e2b106fba8a0b17860a61bc38a3868d38c72e4a4/sqlmesh/core/schema_loader.py#L91)

This is the write path, so existing records with an empty gateway will be untouched unless the user is in [single gateway mode](https://github.com/TobikoData/sqlmesh/blob/a7e75dc98455f1e36a4bb9dde5bd71d0b91c54f9/sqlmesh/core/config/root.py#L89) where the gateway is named `''`and then they will be overwritten.

On the read path, records with no gateway will be loaded and then merged with records for the current gateway based on the `name`. If the resulting model is the wrong one (and only the user can determine this), then the user needs to manually edit the file to set `gateway:` on the record with no gateway and then remove the record with the gateway set.

Alternatively, the user can delete the file entirely and start from scratch which will ensure that all records are recreated with the proper `gateway:` set